### DISCUSSION TITLE
Add authentification profil

### DISCUSSION THREAD

**Discussion by JeanMarieEq** [electerm/electerm#3446] [Ideas]
Hi,
first of all, thank you for this awsome application.

Can you add authentification profil manager in settings ?
Somethings like : 

- profil name
- login
- password

In bookmark creation, add option to select authentification profil rather than enter login and password.

I have to connect on many servers where auth use Active Directory with 1 mounth password renew policy.
So every mounth i spend so much time to change password.

Thank for your concern

**Comment by zxdong262** [comment]
Got it, let me do something about it

**Comment by zxdong262** [comment]
How about a password manager, batch change pass for all bookmarks with same pass?

**Comment by JeanMarieEq** [comment]
Hi,

I've thought of a solution like this but it poses a problem.
If I have to connect to 2 infra which use two different Active directories, how can the batch distinguish between infra 1 and infra 2?

That's why I've come up with a profile system that allows precise identification of bookmarks without the risk of overwriting a password that shouldn't be overwritten.

I used mobaXterm for a while (I no longer have a license) and it offers this type of functionality, which I found very useful.

If I can help in any way, please let me know and thank you for your interest.

**Reply by zxdong262** [reply]
Make sense, would add profile feature, thank you for the feedback

**Comment by zxdong262** [comment]
https://github.com/electerm/electerm/issues/3447

**Comment by zxdong262** [comment]
https://github.com/electerm/electerm/releases/tag/v1.39.31 Released new version with auth profile, again, thank you for the feedback, or I may never think about this feature.

**Comment by JeanMarieEq** [comment]
Hi, there,
I'm back from vacation and I've just seen that you've made a release with profile management.

It's really good, exactly what I was missing in your application.
I tested it and it works well.

Maybe there's just one bug (I haven't really had time to test it yet): when you create a new bookmark, the connection test always returns an error, but if you connect with the same bookmark, it works.

![image](https://github.com/electerm/electerm/assets/139954100/2e3d6bcf-374d-48bc-b7f9-58abbd9a92be)


I think it's just that the test function call doesn't implement profile management properly.

Really great work, thank you very much for this feature which is very useful for me and for others I hope.

Translated with DeepL.com (free version)

**Reply by zxdong262** [reply]
Got it, would fix it in next release ,thank you!
